server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1199,20 +1318,21 @@ public void loadNewTsFile(TsFileResource newTsFileResource)
       boolean isOverlap = false;
       int preIndex = -1, subsequentIndex = sequenceFileList.size();
 
+      List<TsFileResource> sequenceList = new ArrayList<>(sequenceFileList);
```

Comment:
The writeLock is acquired previously, I can't see why you copy it.

Reply:
sequenceFileList is not a list but a tree set, but we want user think it's a list so I didn't change it's name. But this method need use list index so we have to cast it to list.

Reply:
If you look carefully, you will find that the use of the index is not necessary anymore. You can replace the iteration by index with an iteration by the iterator, and the following uses of the index can be replaced by the file you find that should be inserted before.

Reply:
It's related to my work LOAD_FILE and sync module. I will have a look at it, something needs to be modified.

Reply:
I do not read this part of code in detail, but sounds something is not correct... A new UT has cover this method after add timeRangeId?